designer/server/src/test/scala/pl/touk/nussknacker/ui/process/deployment/DeploymentServiceSpec.scala

```diff
@@ -109,10 +115,119 @@ class DeploymentServiceSpec
       processValidatorByProcessingType,
       TestFactory.scenarioResolverByProcessingType,
       listener,
-      scenarioStateTimeout = scenarioStateTimeout
+      scenarioStateTimeout,
+      deploymentCommentSettings
     )
   }
 
+  // TODO: temporary step - we would like to extract the validation and the comment validation tests to external validators
+  private def createDeploymentServiceWithCommentSettings = {
+    val commentSettings = DeploymentCommentSettings.unsafe(".+", Option("sampleComment"))
+    val deploymentServiceWithCommentSettings =
+      createDeploymentService(deploymentCommentSettings = Some(commentSettings))
+    deploymentServiceWithCommentSettings
+  }
+
+  test("should return error when trying to deploy without comment when comment is required") {
+    val deploymentServiceWithCommentSettings: DeploymentService = createDeploymentServiceWithCommentSettings
+
+    val processName: ProcessName = generateProcessName
+    val id                       = prepareProcess(processName).dbioActionValues
+
+    val result = deploymentServiceWithCommentSettings.deployProcessAsync(id, None, None).failed.futureValue
+
+    result shouldBe a[ValidationError]
+    result.getMessage.trim shouldBe "Comment is required."
+
+    eventually {
+      val inProgressActions = actionRepository.getInProgressActionTypes(id.id).dbioActionValues
+      inProgressActions should have size 0
+    }
+  }
+
+  test("should not deploy without comment when comment is required") {
+    val deploymentServiceWithCommentSettings: DeploymentService = createDeploymentServiceWithCommentSettings
+
+    val processName: ProcessName = generateProcessName
+    val id                       = prepareProcess(processName).dbioActionValues
+
+    deploymentServiceWithCommentSettings.deployProcessAsync(id, None, None)
+
+    eventually {
+      deploymentServiceWithCommentSettings
```

Comment:
To avoid code duplication you can extract:
```
val status = deploymentServiceWithCommentSettings
        .getProcessState(id)
        .futureValue
        .status
```